salt/modules/esxdatacenter.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+'''
+Module used to access the esxdatacenter proxy connection methods
+'''
+from __future__ import absolute_import
+
+# Import python libs
+import logging
+import salt.utils
+
+
+log = logging.getLogger(__name__)
+
+__proxyenabled__ = ['esxdatacenter']
+# Define the module's virtual name
+__virtualname__ = 'esxdatacenter'
+
+
+def __virtual__():
+    '''
+    Only work on proxy
+    '''
+    if salt.utils.is_proxy():
+        return __virtualname__
+    return False
```

Comment:
IMHO this is should return an error indicating that it needs to be on a proxy. We can do so as follows: `return (False, 'Must be run on a proxy minion`)`